lxd/metrics/metrics.go

```diff
@@ -24,26 +22,19 @@ func NewMetricSet(labels map[string]string) *MetricSet {
 	return &out
 }
 
-// FilterSamples filters the existing MetricSet using the given permission checker. Samples not containing the "project" label are skipped.
-func (m *MetricSet) FilterSamples(permissionChecker func(entityURL *api.URL) bool) {
+// FilterSamples filters the existing MetricSet using the given permission checker.
```

Comment:
The comment doesn't seem to align with the function signature any more, why is a slice of permissions checkers needed?

Reply:
The samples that don't have any labels (internal server metrics) will be allowed by the `can_view_metrics` on the `server` type. The one with labels will be allowed by having `can_view_metrics` on the `project` type. 
By passing both permission checkers we need to iterate the list of samples once and grant all samples in the final output that are permitted by any of the given `permissionChecks`. 

I have updated the description to make it more clear.